diff --git a/Makefile.am b/Makefile.am
index 9dd4801..440e9e4 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -4,7 +4,7 @@
 ACLOCAL_AMFLAGS = -I m4
 
 ## The order of subdirs is important, don't change without a reason.
-SUBDIRS = gen.tab lib bin doc test
+SUBDIRS = gen.tab lib
 
 EXTRA_DIST = autogen.sh ChangeLog.old \
 	meson.build meson_options.txt bin/meson.build doc/meson.build \
diff --git a/configure.ac b/configure.ac
index 722142e..bbf0fc4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -152,9 +152,5 @@ AC_CONFIG_FILES([fribidi.pc
 		 lib/fribidi-config.h
 		 Makefile
 		 gen.tab/Makefile
-		 lib/Makefile
-		 bin/Makefile
-		 doc/Makefile
-		 test/Makefile
-		 test/unicode-conformance/Makefile])
+		 lib/Makefile])
 AC_OUTPUT
