diff --git a/builds/freetype.mk b/builds/freetype.mk
index 111483069..734c3139a 100644
--- a/builds/freetype.mk
+++ b/builds/freetype.mk
@@ -169,7 +169,7 @@ FT_CFLAGS  = $(CPPFLAGS) \
 
 # Include the `exports' rules file.
 #
-include $(TOP_DIR)/builds/exports.mk
+#include $(TOP_DIR)/builds/exports.mk
 
 
 # Initialize the list of objects.
diff --git a/builds/unix/unix-cc.in b/builds/unix/unix-cc.in
index 5675866ea..5e77e3eff 100644
--- a/builds/unix/unix-cc.in
+++ b/builds/unix/unix-cc.in
@@ -115,7 +115,6 @@ CCexe        := $(CCraw_build)	# used to compile `apinames' only
 #
 LINK_LIBRARY = $(LIBTOOL) --mode=link $(CCraw) -o $@ $(OBJECTS_LIST) \
                           -rpath $(libdir) -version-info $(version_info) \
-                          $(LDFLAGS) -no-undefined \
-                          -export-symbols $(EXPORTS_LIST)
+                          $(LDFLAGS) -no-undefined
 
 # EOF
