diff --git a/src/fcstat.c b/src/fcstat.c
index 5aa1643..2a5ae3b 100644
--- a/src/fcstat.c
+++ b/src/fcstat.c
@@ -398,6 +398,8 @@ FcFStatFs (int fd, FcStatFS *statb)
 	}
 
 	return ret;
+#  elif EMSCRIPTEN
+        return ret;
 #  else
 #    error "BUG: No way to figure out with fstatfs()"
 #  endif
@@ -418,17 +420,24 @@ FcFStatFs (int fd, FcStatFS *statb)
 FcBool
 FcIsFsMmapSafe (int fd)
 {
+#ifdef EMSCRIPTEN
+  return FcFalse;
+#else
     FcStatFS statb;
 
     if (FcFStatFs (fd, &statb) < 0)
 	return FcTrue;
 
     return !statb.is_remote_fs;
+#endif
 }
 
 FcBool
 FcIsFsMtimeBroken (const FcChar8 *dir)
 {
+#ifdef EMSCRIPTEN
+  	return FcFalse;
+#else
     int fd = FcOpen ((const char *) dir, O_RDONLY);
 
     if (fd != -1)
@@ -444,6 +453,7 @@ FcIsFsMtimeBroken (const FcChar8 *dir)
     }
 
     return FcFalse;
+#endif
 }
 
 #define __fcstat__
