diff --git a/configure.ac b/configure.ac
index 7c3aa033..127d17d4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -119,14 +119,15 @@ AC_MSG_RESULT([$hb_os_win32])
 AM_CONDITIONAL(OS_WIN32, test "$hb_os_win32" = "yes")
 
 have_pthread=false
-if test "$hb_os_win32" = no; then
-	AX_PTHREAD([have_pthread=true])
-fi
-if $have_pthread; then
-	AC_DEFINE(HAVE_PTHREAD, 1, [Have POSIX threads])
-fi
-AM_CONDITIONAL(HAVE_PTHREAD, $have_pthread)
-
+#if test "$hb_os_win32" = no; then
+#	AX_PTHREAD([have_pthread=true])
+#fi
+#if $have_pthread; then
+#	AC_DEFINE(HAVE_PTHREAD, 1, [Have POSIX threads])
+#fi
+AM_CONDITIONAL(HAVE_PTHREAD, $have_pthread)
+have_pthread=false
+AC_DEFINE(HAVE_PTHREAD, 0, [Have POSIX threads])
 dnl ==========================================================================
 
 have_ot=true
